src/mediarithmics/plugins/audience-feed-connector/AudienceFeedConnectorBasePlugin.ts

```diff
@@ -57,6 +63,31 @@ abstract class GenericAudienceFeedConnectorBasePlugin<
     return response.data;
   }
 
+  async fetchUserAgentIdentifierRealms(datamartId: string): Promise<Array<UserAgentIdentifierRealmSelectionResource>> {
+    return super
+      .requestGatewayHelper<UserAgentIdentifierRealmSelectionResourcesResponse>(
+        'GET',
+        `${this.outboundPlatformUrl}/v1/datamarts/${datamartId}/user_agent_identifier_realm_selections`,
+      )
+      .then((resRealms) => {
+        this.logger.debug(`Fetched user agent identifier realms for the datamart with id: ${datamartId}`);
+
+        return resRealms.data;
+      });
+  }
+
+  async checkUserAgentIdentifierRealmType(datamartId: string, realmTypeToCheck: RealmType): Promise<boolean> {
+    return this.fetchUserAgentIdentifierRealms(datamartId).then((realms) => {
+      const hasRealm = realms.some((realm) => realm.realm_type === realmTypeToCheck);
+      if (hasRealm === false) {
+        throw new AudienceFeedInstanceContextError(
+          `No user agent identifier realm selection of type ${realmTypeToCheck} was found in datamart ${datamartId}`,
+        );
+      }
```

Comment:
why not just returning false ?

Reply:
Because the spec in the ticket says to throw a public error in this case